corehq/apps/es/case_search.py

```diff
@@ -177,6 +177,29 @@ def _from_dict(self, case):
         doc['_id'] = case_dict['_id']
         return super()._from_dict(doc)
 
+    def _get_domain_from_doc(self, doc):
+        """
+        `doc` can be CommcCareCase instance or dict. This util method extracts domain from doc.
+        This will fail hard if domain is not present in doc.
+        """
+        if isinstance(doc, dict):
+            return doc["domain"]
+        if hasattr(doc, 'domain'):
+            return doc.domain
+
+    def index(self, doc, refresh=False):
+        """
+        Selectively multiplexes writes to a sub index based on the domain of the doc.
+        """
+        adapter = multiplex_to_adapter(self._get_domain_from_doc(doc))
```

Comment:
This is the secondary adapter right?  I'd use some language in here to clarify, since `self` is also an adapter

Reply:
I see elsewhere you called it `sub_index_adapter`, that seems nice and clear - think it'd be appropriate to add it here too?

Reply:
02b8735d9eba66f3849ae96e74ab5807ae959971